skunkworks_crow/src/main/java/org/odk/share/network/WifiHotspotManager.java

```diff
@@ -0,0 +1,81 @@
+package org.odk.share.network;
+
+import android.content.Context;
+import android.location.LocationManager;
+import android.net.wifi.WifiConfiguration;
+import android.os.Build;
+
+import org.odk.share.preferences.SharedPreferencesHelper;
+
+import javax.inject.Inject;
+
+import static android.content.Context.LOCATION_SERVICE;
+
+public final class WifiHotspotManager {
+
+    @Inject
+    Context context;
+    @Inject
+    SharedPreferencesHelper sharedPreferencesHelper;
+    @Inject
+    WifiHospotConnector wifiHotspot;
+
+    @Inject
+    WifiHotspotManager() {
+
+    }
+
+    private boolean isGpsEnabled() {
+        LocationManager locationManager = (LocationManager) context.getSystemService(LOCATION_SERVICE);
+        return locationManager != null && locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+    }
+
+    public void startHotspot() throws GpsNotEnabledException, TriggerHotspotException {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            // check if GPS enabled, if not then show location alert dialog to enable
+            // show system alert dialog to enable wifi tethering
+            if (!isGpsEnabled()) {
+                throw new GpsNotEnabledException();
+            } else {
+                throw new TriggerHotspotException();
+            }
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            // show system alert dialog to enable wifi tethering
+            throw new TriggerHotspotException();
+        } else {
+            // enable wifi tethering
+            wifiHotspot.enableHotspot(sharedPreferencesHelper.getHotspotName());
+        }
+    }
+
+    public void stopHotspot() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
```

Comment:
For those version checks, we can build a boolean variable like `isApiHigherThanO = ...`, can call `isApiHigherThanO` or `!isApiHigherThanO` if necessary, which can simplify the code and make them better to read.